rust/lanzatool/src/install.rs

```diff
@@ -99,10 +100,15 @@ impl Installer {
         // mismatches.
         install(&initrd_location, &esp_paths.initrd).context("Failed to install initrd to ESP")?;
 
+        // TODO: can we avoid this clone here?
+        let secureboot_extensions = bootspec.extensions
+            .clone()
+            .context("Missing required extensions for SecureBoot in Bootspec")?;
```

Comment:
@blitz @nikstur if you have simple ideas to escape from the borrow checker police behind the shared reference of `bootspec` to avoid the clone, I am all ears

Reply:
This should do it: 
```rust
let extensions = &bootspec
     .extensions
     .as_ref()
     .ok_or_else(|| anyhow!("Missing required extensions for SecureBoot in Bootspec"))?;
```

Reply:
Makes sense

Reply:
@nikstur why `.ok_or_else` ? `.context` is implement on `Option<T>` right?

Reply:
Ah I didn't know that. Context is better than.